### PR TITLE
Remove cyclic imports, reduce worker loading time

### LINKED ISSUE #3165: Improve loading time of worker
- Use dynamic imports in worker to load it faster. This is one of the main slowdowns on login as worker is a big chunk and we probably don't need all the indexer things to log in.
- For development we compile to ES6. Dynamic imports are currently not supported in Firefox ~~or Safari~~ supported since Safari 15, Firefox in development. 
- Another idea: Create separate worker thread for search index

## Test notes
 - [ ] Test all platforms
   - [ ] android app
   - [ ] ios app
   - [ ] desktop app
   - [ ] Chrome
   - [ ] Firefox
   - [ ] Chrome on Android
   - [ ] Firefox on Android
   - [ ] Safari desktop
   - [ ] Safari ios

### PR DESCRIPTION
**Pull request by charlag**
close #3165

### CODE REVIEW

File: src/api/common/WorkerProxy.ts [hunk 1]
```diff
@@ -44,6 +44,39 @@ export function exposeLocal<T, IncomingRequestType>(impls: T): (message: Request
...
+		const [facade, fn, args] = message.args
+		const init = downcast(impls)[facade]
+
+		if (init == null) {
+			throw new ProgrammingError(`Facade is not exposed: ${facade}.${fn} (exposeLocal)`)
+		}
+
+		const impl = await init()
+		if (impl == null) {
+			throw new ProgrammingError(`Facade is not lazy: ${facade}.${fn} (exposeLocalDelayed)`)
```
Comment (ganthern) [inline src/api/common/WorkerProxy.ts:73]: I'm pretty sure that an await on a non-promise value returns that value, no?

File: src/api/worker/WorkerLocator.ts [hunk 2]
```diff
@@ -265,26 +274,28 @@ export async function initLocator(worker: WorkerImpl, browserData: BrowserData)
 	)
 	locator.mail = new MailFacade(locator.user, locator.file, locator.cachingEntityClient, locator.crypto, locator.serviceExecutor, locator.blob, fileApp)
 	const nativePushFacade = new NativePushFacadeSendDispatcher(worker)
-	// not needed for admin client
-	if (!isAdminClient()) {
-		locator.calendar = new CalendarFacade(
+	locator.calendar = lazyMemoized(async () => {
```
Comment (ganthern) [inline src/api/worker/WorkerLocator.ts:309]: are we just hoping that the admin client never calls this? i guess that's fair

**Comment by ganthern** [comment]
* test notes should probably be on the issue
* we need to adapt the admin client as well :(
* after this, should we make cyclic imports a hard fail?

File: src/api/common/WorkerProxy.ts (hunk 1 above)
Comment (charlag) [reply src/api/common/WorkerProxy.ts:73]: the promise might resolve to `null` or `undefined` but the error message is wrong then, that's true

File: src/api/worker/WorkerLocator.ts (hunk 2 above)
Comment (charlag) [reply src/api/worker/WorkerLocator.ts:309]: that was the hope before as well except it would explode more explicitly

**Pull request merged**